File: leetcode/1_bit_and_2_bit_characters/javascript/solution.js
class Solution {
   /**
    * Time complexity: O(n)
    * Auxiliary space complexity: O(1)
    * Tags:
    *     A: iteration
    * @param {bits}
    * @return {boolean}
    */
   isOneBitCharacter(bits) {
      let index = 0
      let isOneBit = false;

      while (index < bits.length) {
         if (bits[index]) {
            isOneBit = false;
            index += 2;
         }
         else {
            isOneBit = true;
            index++;
         }
      }

      return isOneBit
   };
}


const isOneBitCharacter = new Solution().isOneBitCharacter;
console.log(new Solution().isOneBitCharacter([1, 0, 0]) === true)
console.log(new Solution().isOneBitCharacter([1, 1, 1, 0]) === false)
console.log(new Solution().isOneBitCharacter([0, 0]) === true)

File: leetcode/1_bit_and_2_bit_characters/python/solution.py
class Solution:
    def isOneBitCharacter(self, bits: list[int]) -> bool:
        """
        Time complexity: O(n)
        Auxiliary space complexity: O(1)
        Tags:
            A: iteration
        """
        N = len(bits)
        index = 0
        is_one_bit = True

        while index < N:
            if bits[index]:
                is_one_bit = False
                index += 2
            else:
                is_one_bit = True
                index += 1

        return is_one_bit


print(Solution().isOneBitCharacter([1, 0, 0]) == True)
print(Solution().isOneBitCharacter([1, 1, 1, 0]) == False)
print(Solution().isOneBitCharacter([0, 0]) == True)
